src/models/src/app_state.rs

```diff
@@ -154,6 +163,52 @@ impl AppState {
             .rp_name(&rp_name);
         let webauthn = Arc::new(builder.build().expect("Invalid configuration"));
 
+        let atproto = {
+            let http_client = Arc::new(DefaultHttpClient::default());
+
+            let listen_scheme = match listen_scheme {
+                ListenScheme::Http | ListenScheme::UnixHttp => "http",
+                ListenScheme::Https | ListenScheme::HttpHttps | ListenScheme::UnixHttps => "https",
+            };
+
+            let client_metadata = AtprotoClientMetadata {
+                client_id: format!(
+                    "{listen_scheme}://{public_url}/auth/v1/atproto/client_metadata"
+                ),
+                client_uri: format!("{listen_scheme}://{public_url}"),
+                redirect_uris: vec![format!(
+                    "{listen_scheme}://{public_url}/auth/v1/atproto/callback"
+                )],
+                token_endpoint_auth_method: AuthMethod::None,
+                grant_types: vec![GrantType::AuthorizationCode],
+                scopes: vec![Scope::Known(KnownScope::Atproto)],
+                jwks_uri: None,
+                token_endpoint_auth_signing_alg: None,
+            };
+
+            Arc::new(
+                OAuthClient::new(OAuthClientConfig {
+                    client_metadata: client_metadata.clone(),
+                    keys: None,
+                    resolver: OAuthResolverConfig {
+                        did_resolver: CommonDidResolver::new(CommonDidResolverConfig {
+                            plc_directory_url: DEFAULT_PLC_DIRECTORY_URL.to_string(),
+                            http_client: http_client.clone(),
+                        }),
+                        handle_resolver: AtprotoHandleResolver::new(AtprotoHandleResolverConfig {
```

Comment:
For everything in this block that's coming from an external dependency, I would like to have one additional module definition added, just to make very clear that it's an external function or struct.

For instance, something like just `AuthMethod` on its own makes sense in many different contexts and by writing it as `atproto::AuthMethod`, it's clear where it belongs to. Another way to solve these issues is to push this initialization into a separate file.

Reply:
Good point, I think it's best to just make the type `AtprotoClient` a newtype struct with a `new` method to avoid the clutter here.